virus-interpreter/src/main/java/com/hartwig/hmftools/virusinterpreter/VirusInterpreterAlgo.java

```diff
@@ -104,6 +109,21 @@ boolean report(VirusBreakend virusBreakend, double expectedClonalCoverage, final
         return reported && virusQCStatus;
     }
 
+    @VisibleForTesting
+    boolean blacklist(int taxidGenus, int taxidSpecies, int taxid, boolean reported)
+    {
+        boolean blacklisted = virusBlacklistingDb.stream()
+                .anyMatch(blacklistedVirus -> taxidGenus == blacklistedVirus.taxid() || taxidSpecies == blacklistedVirus.taxid());
+        if(blacklisted && reported)
+        {
+            throw new RuntimeException("Virus with taxid {} configured as reported and as blacklisted" + taxid);
```

Comment:
I like the exception here! Though it is more typical to throw an `IllegalStateException`

Also, may be an idea to add this check one level higher? In that case you don't have to pass the actual taxid into this function?

Reply:
Thanks, much better indeed!